### ISSUE TITLE
Systemd automount prevents flatpak apps from launching on Ubuntu 17.10

### ISSUE THREAD

**Issue by hackel** [flatpak/flatpak#1121]
I've just noticed this behaviour, which I now believe may  have started after upgrading to Ubuntu 17.10.  When I run a flatpak app, nothing happens.  This may be an incompatibility with 17.10, since I'm still running the old package 0.9.99-0alexlarsson1~zesty.  I don't believe I used to have this problem.
```
$ flatpak run -v com.belledonnecommunications.linphone
XA: Opening user flatpak installation at path /home/hackel/.local/share/flatpak
XA: No installations directory in /etc/flatpak/installations.d. Skipping
XA: Opening user flatpak installation at path /home/hackel/.local/share/flatpak
XA: Regenerating ld.so.cache /home/hackel/.var/app/com.belledonnecommunications.linphone/.ld.so/0d9a4605b0aa65a31c8b2e91c81432511209a59fa83a3a342a2b60b4d4d0f95c
XA: Xdg dir xdg-download is $HOME (i.e. disabled), ignoring
XA: Allowing x11 access
XA: Allowing pulseaudio access
Can't bind mount /oldroot/landru on /newroot/landru: No such device
```
At this point it just sits and does nothing.  It seems as if the app is working, but nothing ever appears on the screen.  I have to Ctrl-C to stop it.  landru is an NFS share on my local network (not currently mounted).  It's managed by systemd:
```
systemd-1 on /landru type autofs (rw,relatime,fd=32,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=20124)
```
Stopping the automount with `systemctl stop landru.automount` allows flatpak apps to successfully run.  I don't know if it's trying to probe all mount points and connect to an inaccessible IP or what.  For now, the workaround is fine since it's not a huge issue to mount it manually when I need it.

**Comment by mwleeds** [comment]
[Here's](https://github.com/projectatomic/bubblewrap/blob/96fee6f4f7b9f4a27a31c0e43a95b8fc513d4ed3/bubblewrap.c#L895) where the error is coming from

**Comment by alexlarsson** [comment]
It seems like this is actually caused by the automount mountpoint. Did you have the same setup before updating to 17.10?

**Comment by alexlarsson** [comment]
Can you please test with flatpak 0.10.0, which is now in the PPA, including for artful. Because i'm not able to reproduce this on Fedora 26:

```
[alex@localhost ~] $ grep testdata /proc/self/mounts
systemd-1 /testdata autofs rw,relatime,fd=30,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=3355684 0 0
[alex@localhost ~] $ flatpak run --command=sh org.gnome.gedit 
sh-4.3$ grep testdata /proc/self/mounts
172.31.0.20:/volume1/data /testdata nfs rw,nosuid,nodev,relatime,vers=3,rsize=131072,wsize=131072,namlen=255,hard,proto=tcp,timeo=600,retrans=2,sec=sys,mountaddr=172.31.0.20,mountvers=3,mountport=892,mountproto=udp,local_lock=none,addr=172.31.0.20 0 0
```

This seems like an equivalent setup, no?

**Comment by gholmlund** [comment]
I am running a fresh install of ubuntu 17.10 and I see the same problem as described above. ( Can't bind mount /oldroot/users .... ). After disabling autofs, it works correctly. I am hoping for an answer that allow me to continue using autofs. My autofs directory ( /os ) is not related to any system directories.

**Comment by gholmlund** [comment]
Actually I have a second autofs directory ( /users ). If I comment it out, I can enable autofs and flatpak is working correctly. Does this problem related specifically to /users?

**Comment by alexlarsson** [comment]
+gholmlund I assume your $HOME is under /users? Maybe that is what makes the difference between /os and /users.

**Comment by gholmlund** [comment]
No. My $HOME is under /home. The /users is a smb mount of a remote directory.

**Comment by Mikenux** [comment]
@hackel @gholmlund: Is this still reproducible?

**Comment by hackel** [comment]
@Mikenux I'm sorry, I no longer have this NFS setup to test against.

**Comment by Mikenux** [comment]
@hackel: Thanks for responding! Let's see if others still have the problem then.
